### PR TITLE
feat: implement public squad submit for review modal

### PR DESCRIPTION
**Pull request by kopancek**
## Changes

- Added modal for submitting public squad for review.
- Right now the submit button is not doing anything, as is specified in the task itself:
https://dailydotdev.atlassian.net/browse/MI-339
- Tidy up the icon mess that I created in the PR for adding eligibility card

Figma link:
https://www.figma.com/file/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=2537-20090&mode=dev

### Screenshots

Screenshot 2024-05-13 at 15 12 11
Screenshot 2024-05-13 at 15 21 44

Screenshot 2024-05-13 at 15 12 19
Screenshot 2024-05-13 at 15 13 24

### On those affected packages:
- [X] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [X] MobileL (420px)
- [X] Tablet (656px)
- [X] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-339 #done

### Preview domain
https://mi-339-submit-for-review-modal.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/icons/index.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ export * from './Card';
 export * from './CardLayout';
 export * from './ChecklistA';
 export * from './ChecklistB';
-export * from './ClockPending';
```
Comment (rebelchris) [inline packages/shared/src/components/icons/index.ts:19]: What was this before?

File: packages/shared/src/components/modals/squads/SubmitSquadForReviewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        <SubHeader>
+          Follow the checklist below to increase the chances of passing the
+          review successfully
+        </SubHeader>
+        <Checklist>
+          <ChecklistItem iconTag={MegaphoneIcon}>
+            The posts I shared meet the{' '}
+            <a
+              className="text-text-link underline hover:no-underline"
+              href="https://docs.daily.dev/docs/for-content-creators/content-guidelines"
```
Comment (rebelchris) [inline packages/shared/src/components/modals/squads/SubmitSquadForReviewModal.tsx:83]: I assume this should be targeting blank? (but maybe double check)

File: packages/shared/src/components/modals/squads/SubmitSquadForReviewModal.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        secondary
+      />
+      <span className="flex-1">{children}</span>
+    </li>
+  );
+};
+
+const Checklist = classed(
+  'ul',
+  'flex flex-col gap-5 p-5 -my-2.5 rounded-16 bg-theme-float typo-callout',
```
Comment (rebelchris) [inline packages/shared/src/components/modals/squads/SubmitSquadForReviewModal.tsx:33]: Personally don't like this as it looks like a bug (raised on slack)

**Review by rebelchris — CHANGES_REQUESTED** [review]
Looks pretty good to me
Just the link should probably be blank and have rels if this the case.


Also maybe check if there should be a tooltip on the disabled button (it's something we used to do a lot)

**Review by nensidosari — APPROVED** [review]
Looks good

**Review by rebelchris — APPROVED** [review]

**Pull request merged**